hathor/transaction/transaction.py

```diff
@@ -305,7 +306,12 @@ def _get_token_info_from_inputs(self) -> dict[TokenUid, TokenInfo]:
 
         for tx_input in self.inputs:
             spent_tx = self.get_spent_tx(tx_input)
-            spent_output = spent_tx.outputs[tx_input.index]
+            try:
+                spent_output = spent_tx.outputs[tx_input.index]
+            except IndexError:
+                raise InexistentInput(
```

Comment:
It's just that before, this method was only called during verification. Now, it's called before verification (when retrieving the verification dependencies). So, if the tx is invalid, it will raise this exception during that retrieval.